src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java

```diff
@@ -110,7 +110,11 @@ private void checkOverloadMethodsGrouping(DetailAST objectBlock) {
                 final String methodName =
                         currentToken.findFirstToken(TokenTypes.IDENT).getText();
                 final Integer previousIndex = methodIndexMap.get(methodName);
-                if (previousIndex != null && currentIndex - previousIndex > allowedDistance) {
+                final DetailAST previousSibling = currentToken.getPreviousSibling();
+                final boolean isMethod = previousSibling.getType() == TokenTypes.METHOD_DEF;
```

Comment:
What if previous is a comment? Will this cause a conflict?

It doesn't look like it because this check isn't comment aware. However, add some examples of this for proof (IE: overloaded methods right next to each other but a comment between them).

Reply:
This checks is not comment aware, so comments are ignored completely.

Reply:
> What if previous is a comment? Will this cause a conflict?

No. The check does not care about comments. I have updated the input files and added this test case:

https://github.com/checkstyle/checkstyle/blob/24ab0e93ad43a656fca70f6a7fd16b9d559bafdf/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/InputOverloadMethodsDeclarationOrder.java#L107-L116

Also mentioned this in the documentation examples:

https://github.com/checkstyle/checkstyle/blob/24ab0e93ad43a656fca70f6a7fd16b9d559bafdf/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/Example1.txt#L10-L16